queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py

```diff
@@ -0,0 +1,188 @@
+# Copyright © 2020 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Manages the parties and party roles for a LegalEntity."""
+from __future__ import annotations
+
+import datetime
+import dpath
+from typing import Dict, List, Optional, Tuple
+
+from business_model import db
+from business_model import AlternateName
+from business_model import Filing
+from business_model import LegalEntity
+
+from entity_filer import db
+from entity_filer.exceptions import DefaultException
+from entity_filer.filing_meta import FilingMeta
+from entity_filer.filing_processors.filing_components import legal_entity_info
+from entity_filer.filing_processors.filing_components.parties import get_or_create_party
+from entity_filer.utils.legislation_datetime import LegislationDatetime
+
+
+def update_partner_change(
+        legal_entity: LegalEntity,
+        filing_type: str,
+        change_filing_rec: Filing,
+        change_filing: Dict,
+        filing_meta: Dict,
+):
+    name_request = dpath.util.get(change_filing, f"/{filing_type}/nameRequest", default=None)
+    if name_request and (to_legal_name := name_request.get("legalName")):
+        alternate_name = AlternateName.find_by_identifier(legal_entity.identifier)
+        parties_dict = dpath.util.get(change_filing, f"/{filing_type}/parties")
+
+        legal_entity.legal_name = get_partnership_name(parties_dict)
+
+        legal_entity.alternate_names.remove(alternate_name)
+        alternate_name.end_date = change_filing_rec.effective_date
+        alternate_name.change_filing_id = change_filing_rec.id
+        # alternate_name.delete()
+        db.session.add(alternate_name)
+        db.session.commit()
+        db.session.delete(alternate_name)
+        db.session.commit()
+
+
+        new_alternate_name = AlternateName(
+            bn15=alternate_name.bn15,
+            change_filing_id=change_filing_rec.id,
+            end_date=None,
+            identifier=legal_entity.identifier,
+            name=to_legal_name,
+            name_type=AlternateName.NameType.OPERATING,
+            start_date=alternate_name.start_date,
```

Comment:
this should be same as `update_proprietor_change`

Reply:
updated